core/object/object.cpp

```diff
@@ -1163,10 +1163,31 @@ Error Object::emit_signalp(const StringName &p_name, const Variant **p_args, int
 		uint32_t idx = 0;
 		for (const KeyValue<Callable, SignalData::Slot> &slot_kv : s->slot_map) {
 			slot_conns[idx++] = slot_kv.value.conn;
+
+			bool disconnect = slot_kv.value.conn.flags & CONNECT_ONE_SHOT;
+#ifdef TOOLS_ENABLED
+			if (disconnect && (slot_kv.value.conn.flags & CONNECT_PERSIST) && Engine::get_singleton()->is_editor_hint()) {
+				// This signal was connected from the editor, and is being edited. Just don't disconnect for now.
+				disconnect = false;
+			}
+#endif
+			if (disconnect) {
+				_ObjectSignalDisconnectData dd;
+				dd.signal = p_name;
+				dd.callable = slot_kv.value.conn.callable;
+				disconnect_data.push_back(dd);
+			}
 		}
 		DEV_ASSERT(idx == s->slot_map.size());
 	}
 
+	while (!disconnect_data.is_empty()) {
```

Comment:
Sorry for some reason this comment was erased from the review. I meant that since the for loop below iterates over a copy anyway, there is no need to make the disconnect_data list anyway, could do just simply before calling in the loop below.

Reply:
er could do the erasing, sorry